Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -459,8 +462,35 @@ def _addToLog(self, logName, message):
     def evaluateCommand(self, cmd):
         rc = super().evaluateCommand(cmd)
         steps_to_add = self.follow_up_steps()
-        if rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash'):
-            steps_to_add += [ArchiveMinifiedBuiltProduct(), UploadMinifiedBuiltProduct(), TransferToS3(terminate_build=True)]
+
+        triggers = self.getProperty('triggers', None)
+        if triggers:
+            # Build archive
+            steps_to_add += [ArchiveBuiltProduct()]
+            if CURRENT_HOSTNAME in BUILD_WEBKIT_HOSTNAMES + TESTING_ENVIRONMENT_HOSTNAMES:
+                steps_to_add.extend([
+                    GenerateS3URL(f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}"),
+                    UploadFileToS3(f"WebKitBuild/{self.getProperty('configuration')}.zip", links={self.name: 'Archive'}),
+                ])
+            else:
+                # S3 might not be configured on local instances, achieve similar functionality without S3.
+                steps_to_add.extend([UploadBuiltProduct(), TransferToS3(terminate_build=True)])
+
+            # Minified build archive
+            full_platform = self.getProperty('fullPlatform')
+            if (full_platform.startswith('mac') or full_platform.startswith('ios-simulator') or full_platform.startswith('tvos-simulator')
+               or full_platform.startswith('watchos-simulator')) or (rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash')):
+
+                steps_to_add += [ArchiveMinifiedBuiltProduct()]
+                if CURRENT_HOSTNAME in BUILD_WEBKIT_HOSTNAMES + TESTING_ENVIRONMENT_HOSTNAMES:
+                    steps_to_add.extend([
+                        GenerateS3URL(f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}", minified=True),
+                        UploadFileToS3(f"WebKitBuild/minified-{self.getProperty('configuration')}.zip", links={self.name: 'Minified Archive'}),
+                    ])
+                else:
+                    # S3 might not be configured on local instances, achieve similar functionality without S3.
+                    steps_to_add.extend([UploadMinifiedBuiltProduct(), TransferToS3(terminate_build=True)])
```

Comment:
Ditto. If S3 is not configured, then we shouldn't run TransferToS3() step either. We should remove TransferToS3 from here.